src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -193,6 +194,29 @@ public static Result canEntitySpawn(EntityLiving entity, World world, float x, f
         return event.getResult();
     }
 
+    public static Result canEntitySpawn(EntityLiving entity, World world, float x, float y, float z, MobSpawnerBaseLogic spawner)
+    {
+        if (entity == null)
+            return Result.DEFAULT;
+        LivingSpawnEvent.CheckSpawn event = new LivingSpawnEvent.CheckSpawn(entity, world, x, y, z, spawner);
+        MinecraftForge.EVENT_BUS.post(event);
+        return event.getResult();
+    }
+
+    public static boolean canEntitySpawnSpawner(EntityLiving entity, World world, float x, float y, float z, MobSpawnerBaseLogic spawner)
+    {
+        Result result = canEntitySpawn(entity, world, x, y, z, spawner);
+        if (result == Result.DEFAULT)
+        {
+            return entity.getCanSpawnHere() && entity.isNotColliding(); // vanilla logic
+        }
+        else
+        {
+            return result == Result.ALLOW;
+        }
+    }
+
+    @Deprecated
```

Comment:
This should be
`@Deprecated // TODO remove in 1.13. use the MobSpawnerBaseLogic version above.`

Reply:
Can put the suggested replacement in a `@deprecated` javadoc tag.